src/ast/nodes/ArrowFunctionExpression.ts

```diff
@@ -36,12 +36,15 @@ export default class ArrowFunctionExpression extends FunctionBase {
 		interaction: NodeInteraction,
 		context: HasEffectsContext
 	): boolean {
-		if (super.hasEffectsOnInteractionAtPath(path, interaction, context)) return true;
```

Comment:
I am not convinced this should take lower priority. This basically covers that arrow.foo.bar is a side effect because it can throw an error, and it would also cover cases where a side-effectful getter is attached as a property to an arrow function. Suppressing their side effects via this annotation feels unexpected to me.